grpc/shared_conn.go

```diff
@@ -301,7 +301,7 @@ func (sc *SharedConn) Close() error {
 	var err error
 	sc.peerConnMu.Lock()
 	if sc.peerConn != nil {
-		err = sc.peerConn.Close()
+		err = sc.peerConn.GracefulClose()
 		// `PeerConnection.Close` returning does not guarantee that background workers have
```

Comment:
This comment seems no longer relevant, and it makes me wonder whether the `select` below is necessary.

Reply:
Thanks for finding these -- I definitely wasn't looking close enough